pkg/ioutils/lazy_reader_disk.go

```diff
@@ -163,36 +164,38 @@ func (r *diskBackedLazyReaderAt) ensureOverflowToDisk(till int64) {
 		r.pos = r.maxBufferSize
 	}
 
-	if r.overFlowFile == nil {
+	if r.overflowFile == nil {
 		var err error
+		// "" indicates we want to use os.TempDir().
 		r.dirPath, err = os.MkdirTemp("", tmpDirName)
 		if err != nil {
 			r.err = errors.Wrap(err, "failed to create temp dir for overflow")
 			return
 		}
 		defer func() {
-			if r.overFlowFile == nil {
+			if r.overflowFile == nil {
 				_ = os.RemoveAll(r.dirPath)
 				r.dirPath = ""
 			}
 		}()
 
 		// Prepare overFlowFile
 		filePath := filepath.Join(r.dirPath, tmpFileName)
-		r.overFlowFile, err = os.Create(filePath)
+		r.overflowFile, err = os.Create(filePath)
 		if err != nil {
 			r.err = errors.Wrapf(err, "create overFlowFile %s", filePath)
 			return
 		}
 	}
 
-	// Copy until next block align with size r.maxBufferSize or the size of file.
+	// Copy until next block align with size overflowBlockSize or the size of file.
 	// Request to copy an extra byte to ensure EOF is recorded.
-	to := mathutil.MinInt64(((till-1)/r.maxBufferSize+1)*overflowBlockSize, r.size)
+	to := mathutil.MinInt64(((till-1)/overflowBlockSize+1)*overflowBlockSize, r.size)
```

Comment:
`r.maxBufferSize` was the original block size, but it was switched over to `overflowBlockSize`; however, this division was missed.